src/System.Collections/tests/Performance/Perf.Dictionary.cs

```diff
@@ -193,5 +193,356 @@ public void ContainsKey(int size)
                         dict.ContainsKey(key); dict.ContainsKey(key); dict.ContainsKey(key);
                     }
         }
+
+        [Benchmark]
+        [InlineData(1)]
+        [InlineData(10)]
+        [InlineData(1000)]
+        public static void Remove_ValueType(long size)
+        {
+            s_collection1 = new Dictionary<long?, long?>();
+            long?[] items;
+
+            items = new long?[size * 10];
+
+            for (long i = 0; i < size * 10; ++i)
+            {
+                items[i] = i;
+                s_collection1.Add(items[i], items[i]);
+            }
+
+
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (long i = 1; i < size; ++i)
+                        s_collection1.Remove(items[i]);
+        }
+
+        [Benchmark]
+        public static void Indexer_get_ValueType()
+        {
+            int size = 1024;
+
+            s_collection = new Dictionary<int?, int?>();
+            for (int i = 0; i < size; ++i)
+            {
+                s_collection.Add(i, i);
+            }
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int j = 0; j < size; ++j)
+                    {
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                        s_item = (int)s_collection[j];
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        public static void Enumeration_ValueType()
+        {
+            int size = 1024;
+            s_collection = new Dictionary<int?, int?>();
+
+            for (int i = 0; i < size; ++i)
+            {
+                s_collection.Add(i, i);
+            }
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    foreach (KeyValuePair<int?, int?> tempItem in s_collection)
+                    {
+                        s_key = tempItem.Key;
+                        s_value = tempItem.Value;
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        public static void Add_ValueType()
```

Comment:
Duplicate of https://github.com/dotnet/corefx/pull/7152/files#diff-53f002bec73f1f8b96c66bcef4e13b18R33 ?